scipy/stats/mstats_basic.py

```diff
@@ -382,9 +382,10 @@ def pearsonr(x,y):
     df = n-2
     if df < 0:
         return (masked, masked)
-
-    (mx, my) = (x.mean(), y.mean())
-    (xm, ym) = (x-mx, y-my)
+        
+    # Mask arrays to make same size
+    (xm, ym) = (ma.MaskedArray(x, m), ma.MaskedArray(y, m))
```

Comment:
You reassign to (xm, ym) in the next line (388) based on the original x, y
I guess you meant to use xm and ym  in line 388